### DISCUSSION TITLE
Distinct Filter

### DISCUSSION THREAD

**Discussion by TahLoow** [pocketbase/pocketbase#1686] [Q&A] [answered]
Hello,

I'm looking for a way to "`SELECT DISTINCT`" values in a field via a filter/SDK. There may be tens of thousands of records in my target collection. 

I searched for the keyword "distinct" here and in the roadmap. [This post](https://github.com/pocketbase/pocketbase/discussions/962) used DISTINCT in a Go hook. We've been hesitant to adopt the Go framework due to learning curves and maintenance overhead. Is a Go app the only option beyond fetching and iterating through all records?

As others have said, your efforts deserve bounties of kudos.

**Comment by ganigeorgiev** [comment, marked as answer]
`DISTINCT` is already applied by default to the list/search query when there are joins and you should never have duplicated records in the result set.

If you want to apply `DISTINCT/GROUP BY` only by a specific single field, then this is not possible at the moment without defining a [custom endpoint](https://pocketbase.io/docs/router/) and [query](https://pocketbase.io/docs/database/).

There are plans in the future to allow creating custom queries from the Admin UI, either as a special field and/or new "view" collection type (_I'll consider prioritizing it after v0.13/v0.14 since other users also requested it_).

**Comment by AlejandroAkbal** [comment]
Do we know if this is currently possible without creating a custom endpoint?

**Reply by ganigeorgiev** [reply]
This is a very old discussion and I'm not sure that I understand what do you refer to.

In general if you want to specify a custom SELECT query, in the latest PocketBase version you can define a [View Collection](https://pocketbase.io/docs/collections/#view-collection) (_Admin UI > + New collection > Type: View_).

If that's not what you are looking for, please open a new discussion with more details about your use case.
